src/plugins/walletConnectToDapps/components/WalletConnectSigningModal/content/TransactionContent.tsx

```diff
@@ -48,14 +51,23 @@ export const TransactionContent: FC<TransactionContentProps> = ({ transaction, c
 
   const { simulationQuery } = useSimulateEvmTransaction({ transaction, chainId, speed })
 
+  const [cachedSimulationData, setCachedSimulationData] =
+    useState<TenderlySimulationResponse | null>(null)
```

Comment:
Once we have a sim, we have a sim - since this is a component instance for a given `transaction`, then effectively simulated data never changes as it's deterministically the same regardless of speed (gas params).